rskj-core/src/main/java/co/rsk/cli/CliToolRskContextAware.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2021 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package co.rsk.cli;
+
+import co.rsk.RskContext;
+import co.rsk.util.Factory;
+import co.rsk.util.NodeStopper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import java.util.Objects;
+
+/**
+ * An abstract class for cli tools that need {@link RskContext}. Lifecycle of the  {@link RskContext} instance
+ * is being managed by {@link CliToolRskContextAware}.
+ *
+ * Also {@link CliToolRskContextAware} provides a logger instance for derived classes.
+ */
+public abstract class CliToolRskContextAware {
```

Comment:
More of a question than a comment: why choose this as a sort of wrapper class? Wouldn't it be easier for CLI tools to just inherit from this class? `execute` would work mostly the same, calling an abstract method that must be implemented.